homeassistant/components/matter/entity.py

```diff
@@ -62,6 +62,8 @@ class MatterEntityDescription(EntityDescription):
     measurement_to_ha: Callable[[Any], Any] | None = None
     ha_to_native_value: Callable[[Any], Any] | None = None
     command_timeout: int | None = None
+    # List of Matter FixedLabel or UserLabels used for name
+    name_using_matter_labels: list | None = None
```

Comment:
why is this a list and if it needs to be a list, the typing is not complete

Reply:
Its a list of strings. Previously, there was a hard-coded list of two labels ("Label" and "Button"), but there are no standards for how manufacturers should label things. In this iteration, I wanted to have a more adaptable approach, and this allows adapting the list of labels to the device by specifying a list in the discovery schema (which seems less error-prone when it gets changed than changing the main code).